src/adapters/fabric/e2eUtils.js

```diff
@@ -859,13 +861,16 @@ async function invokebycontext(context, id, version, args, timeout){
                 //invokeStatus.Set('invokeLatency', (Date.now() - beforeInvokeTime));
                 one_good = true;
             } else {
-                let err = new Error('Endorsement denied: ' + proposal_response.toString());
+                //let err = new Error('Endorsement denied: ' + proposal_response.toString());
                 errFlag |= TxErrorEnum.BadProposalResponseError;
                 invokeStatus.SetFlag(errFlag);
-                invokeStatus.SetErrMsg(TxErrorIndex.BadProposalResponseError, err.toString());
+                invokeStatus.SetErrMsg(TxErrorIndex.BadProposalResponseError, 'Endorsement denied: ' + proposal_response.toString());
                 // explicit rejection, early life-cycle termination, definitely failed
                 invokeStatus.SetVerification(true);
-                throw err;
+                commLogger.error('Failed to complete transaction [' + txId.substring(0, 5) + '...]:' + ' Endorsement denied: ' + proposal_response.toString());
```

Comment:
The exception is thrown "to the end", so the errors can be handled at one place at the end of the function. Otherwise, there will be a lot of repetitive code.